test/unit/notifications.server.test.ts

```diff
@@ -0,0 +1,40 @@
+import sendNotification from '../../app/lib/notifications.server';
+
+const { NOTIFICATIONS_EMAIL_USER } = process.env;
+
+describe('Notification server lib function test', () => {
+  const recipient = 'dev@starchart.invalid';
+  const subject = 'Sample subject';
+  const text = 'Sample text';
+
+  test('sending and receiving email notification using sendNotification()', async () => {
+    const sentMailResponse = await sendNotification(recipient, subject, text);
+
+    let messageId = sentMailResponse?.messageId;
+
+    expect(sentMailResponse).toEqual(
+      expect.objectContaining({
+        accepted: expect.arrayContaining([recipient]),
+      })
+    );
+
+    const mailhogResponse = await fetch(
+      `http://localhost:8025/api/v2/search?kind=to&query=${recipient}`
+    );
+    const responseToJSON = await mailhogResponse.json();
+    expect(responseToJSON.items).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          Content: expect.objectContaining({
+            Headers: expect.objectContaining({
+              'Message-ID': expect.arrayContaining([messageId]),
+              'To': expect.arrayContaining([recipient]),
+              'From': expect.arrayContaining([NOTIFICATIONS_EMAIL_USER]),
```

Comment:
Not sure if I like how deeply nested this is. Or understand whats going on here....

Reply:
I think this is testing if the response items array contains the object defined here

Reply:
Maybe this could be moved into a variable for better readability?

Reply:
I think it might be nicer to split up the obj, something like this:

```ts
expect(responseToJSON.items.Content).ToContain(...)
expect(responseToJSON.items.Content.Headers).ToContain(...)
```
idk if this is fully possible

Reply:
No, this is pretty standard.  Adding variables adds places where this can fail due to variability.  What's here is looking good to me.

Reply:
Testing each of the objects fields one at a time. Rather than all at once

Reply:
fair enough